File: Medusa/MedusaCore/Core/Config/UpdateServer.cpp
// Copyright (c) 2015 fjz13. All rights reserved.
// Use of this source code is governed by a MIT-style
// license that can be found in the LICENSE file.
#include "MedusaCorePreCompiled.h"
//SIREN_BODY_INCLUDE_BEGIN
#include "UpdateServer.h"
//SIREN_BODY_INCLUDE_END

MEDUSA_BEGIN;

UpdateServer::UpdateServer()
{
//SIREN_BODY_CONSTRUCT_BEGIN
	mMinAppVersion = 0;
	mCurFileVersion = 0;
	mStatus = (UpdateServerStatus)0;
//SIREN_BODY_CONSTRUCT_END
}

UpdateServer::~UpdateServer()
{
//SIREN_BODY_DESTRUCT_BEGIN
//SIREN_BODY_DESTRUCT_END
}

//SIREN_BODY_METADATA_BEGIN
SIREN_METADATA(UpdateServer, 12);
SIREN_PROPERTY_METADATA(0, UpdateServer, MinAppVersion, 13, 0);
SIREN_PROPERTY_METADATA(1, UpdateServer, CurFileVersion, 14, 0);
SIREN_PROPERTY_METADATA_STRUCT(2, UpdateServer, Description, 11);
SIREN_PROPERTY_METADATA(3, UpdateServer, Status, 6, (UpdateServerStatus)0);
SIREN_PROPERTY_METADATA_STRUCT(4, UpdateServer, Patches, 7);
SIREN_PROPERTY_METADATA_STRUCT(5, UpdateServer, JumpPatches, 11);
//SIREN_BODY_METADATA_END

MEDUSA_END;

File: Medusa/MedusaProto/Core/UpdateServer.cs
// Copyright (c) 2015 fjz13. All rights reserved.
// Use of this source code is governed by a MIT-style
// license that can be found in the LICENSE file.
using System.Collections.Generic;
using Siren;
using Siren.Attribute;

namespace Medusa.Core
{
    [SirenClass(typeof(MedusaCoreTemplate), "Core/Config")]
    public enum UpdateServerStatus
    {
        OK = 0,
        Maintain = 1,
        Error = 2,
        DontUpgrade = 3
    };

    [SirenClass(typeof (MedusaCoreTemplate), "Core/Config", SirenGenerateMode.SirenConfig)]
    public class PackageInfo
    {
        [SirenProperty]
        public string Description { get; set; }

        [SirenProperty]
        public string Path { get; set; }

        [SirenProperty]
        public string Signature { get; set; }   //used to check after download from internet
    }

    [SirenClass(typeof(MedusaCoreTemplate), "Core/Config", SirenGenerateMode.SirenConfig)]
    public class UpdateServer
    {
        [SirenProperty]
        public uint MinAppVersion { get; set; }

        [SirenProperty]
        public uint CurFileVersion { get; set; }

        [SirenProperty]
        public string Description { get; set; }

        [SirenProperty]
        public UpdateServerStatus Status { get; set; }

        [SirenProperty]
        public Dictionary<uint, PackageInfo> Patches { get; set; }

        [SirenProperty]
        public Dictionary<uint, PackageInfo> JumpPatches { get; set; }
    }
}
